src/core/scene.cpp

```diff
@@ -1,394 +1,421 @@
-    }
-    mNewActors.clear();
-}
-
-ActorList_t const& Scene::GetActors() const
-{
-    return mActorHolder.mAllActors;
-}
-
-ActorList_t& Scene::GetActors()
-{
-    return mActorHolder.mAllActors;
-}
-
-void Scene::SetActors(ActorList_t& actors, bool withAddActorEvents/*=true*/)
-{
-    for( NewActorList_t::iterator it = mNewActors.begin(), e = mNewActors.end(); it != e; ++it )
-    {
-        mActorHolder.mAllActors.insert( *it );
-    }
-    mNewActors.clear();
-
-    for( ActorList_t::iterator it = mActorHolder.mAllActors.begin(), e = mActorHolder.mAllActors.end(); it!=e; ++it )
-    {
-        delete (*it).Get();
-    }
-    mActorHolder.mAllActors.clear();
-    mActorHolder.mAllActors = actors;
-    if (withAddActorEvents)
-    {
-        for( ActorList_t::iterator it = mActorHolder.mAllActors.begin(), e = mActorHolder.mAllActors.end(); it!=e; ++it )
-        {
-            EventServer<ActorEvent>::Get().SendEvent( ActorEvent( (*it), ActorEvent::Added ) );
-        }
-    }
-}
-
-void Scene::ClearActors( bool withEvents/*=true*/ )
-{
-    mPaused = false;
-    
-    for( NewActorList_t::iterator it = mNewActors.begin(), e = mNewActors.end(); it != e; ++it )
-    {
-        if (withEvents)
-        {
-            EventServer<ActorEvent>::Get().SendEvent( ActorEvent( (*it), ActorEvent::Added ) );
-        }
-        L2("new actor inserted at Clear (GUID:%d)\n",(*it)->GetGUID());
-        mActorHolder.mAllActors.insert( *it );
-    }
-    mNewActors.clear();
-
-    for( ActorList_t::iterator it = mActorHolder.mAllActors.begin(), e = mActorHolder.mAllActors.end(); it!=e; ++it )
-    {
-        if (withEvents)
-        {
-            EventServer<ActorEvent>::Get().SendEvent( ActorEvent( (*it), ActorEvent::Removed ) );
-        }
-        L2("actor deleted at Clear (GUID:%d)\n",(*it)->GetGUID());
-        delete (*it).Get();
-    }
-    mActorHolder.mAllActors.clear();
-}
-
+    if (!actor.IsValid())
+    {
+        return;
+    }
+    mPlayerModels.push_back( new ModelValue( (ModelValue::get_int_t) boost::lambda::bind( &getHP, actor.Get() ), "hp", &mPlayerModel ) );
+    mPlayerModels.push_back( new ModelValue( (ModelValue::get_double_t) boost::lambda::bind( &getX, actor.Get() ), "x", &mPlayerModel ) );
+    mPlayerModels.push_back( new ModelValue( (ModelValue::get_double_t) boost::lambda::bind( &getY, actor.Get() ), "y", &mPlayerModel ) );
+    mPlayerModels.push_back( new ModelValue( (ModelValue::get_int_t) boost::lambda::bind( &getWeaponId, actor.Get() ), "weapon", &mPlayerModel ) );
+    mPlayerModels.push_back( new ModelValue( (ModelValue::get_int_t) boost::lambda::bind( &getSpecialId, actor.Get() ), "special", &mPlayerModel ) );
+    mPlayerModels.push_back( new ModelValue( (ModelValue::get_int_vec_t) boost::lambda::bind( &getBuffs, actor.Get() ), "buffs", &mPlayerModel ) );
+    mPlayerModels.push_back( new ModelValue( RefTo( mMaxHP ), "max_hp", &mPlayerModel ) );
+}
+
+void Scene::SelectLevel(std::string const& Level)
```

Comment:
m: scene should be dumb. Currently it's function is to handle level load, and actors.
this way mSelectedGameMode should be in ProgramState.
Load ui should be handled differently. Currently you can extend to a new game mode without editing too much - or any - core functionality. (more or less, that's the goal that's for sure :) ). There is a capture_the_flag_system and a free_for_all_system i think. There you should handle ui.loads. Thank you bro.